pkg/controller/vsphere/reconciler.go

```diff
@@ -616,7 +617,13 @@ func getDiskSpec(s *machineScope, devices object.VirtualDeviceList) (types.BaseV
 	}
 
 	disk := disks[0].(*types.VirtualDisk)
-	disk.CapacityInKB = int64(s.providerSpec.DiskGiB) * 1024 * 1024
+	cloneCapacityKB := int64(s.providerSpec.DiskGiB) * 1024 * 1024
+	if disk.CapacityInKB > cloneCapacityKB {
+		return nil, machineapierros.InvalidMachineConfiguration(
+			"can't resize template disk down, initial capacity is larger: %dKB > %dKB",
```

Comment:
I think the values here are  in `KiB` as opposed to `KB`, we should try to make sure the error message has the correct units

Reply:
Good catch, will fix that, but keep in mind it is an upstream issue for storage size pedantic people 
![image](https://user-images.githubusercontent.com/32226600/92902915-f7665b80-f421-11ea-8deb-1f2f33300883.png)
This is what vmware UI shows on the template. We should actually multiply by `1000` and call it `GB/MB/KB` etc.